fluent-syntax/src/parser.js

```diff
@@ -2,7 +2,20 @@
 
 import * as AST from './ast';
 import { FTLParserStream } from './ftlstream';
-import { error, getErrorSlice } from './errors';
+import { error } from './errors';
+
+export function lineOffset(source, pos) {
+  // Substract 1 to get the offset.
+  return source.substring(0, pos).split('\n').length - 1;
+}
+
+export function columnOffset(source, pos) {
+  const lastLineBreak = source.lastIndexOf('\n', pos);
+  return lastLineBreak === -1
+    ? pos
+    // Substracting two offsets gives length; substract 1 to get the offset.
+    : pos - lastLineBreak - 1;
+}
```

Comment:
https://github.com/Pike/compare-locales/blob/master/compare_locales/parser.py#L187 is how I code this problem in compare-locales.

Effectively, I cache the position of line breaks in the source, and then use bisect to find which of those positions matter for my current pos. Based on the position of the found line break in the list I know the line number, based on the value, I can compute the column.

I recognize that this is a lot easier to implement in python, as there's no finditer in js regexp :-(